### ISSUE TITLE
I managed to get a reproducible save game before a land unit can embark on -and kill- an already embarked civilian enemy unit

### ISSUE THREAD

**Issue by SuperNoobCamper** [LoneGazebo/Community-Patch-DLL#11166]
_1. Mod Version (X.Y.Z). Current Version: 4.14.0_

![Screenshot (123)](https://github.com/user-attachments/assets/41eaa01f-ac4a-49da-af20-5d59d0a066d5)

_2. DirectX Version_

_3. List of Other Mods_

![Screenshot (121)](https://github.com/user-attachments/assets/9d3cd07f-ea90-490e-a6ad-5aa0f5db053e)

_4. Describe the Issue_

 A land unit should not be able to embark on a tile occupied by an enemy civilian unit, this is a very old -but very inconsistent and hard to repro bug because it usually happens on AI turn and saving & reloading seems to change AI behavior even if i do the exact same moves on my turn- however this particular save is reproducible on human turn as you can see on the attached screenshots; if you order the knight to embark on the highlighted tile it will need two turns to do so, so it ends it's turn on top of an enemy worker and kills it in the process

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_

[ReporJayavarman VII_0277 AD-1285.zip](https://github.com/user-attachments/files/16321603/ReporJayavarman.VII_0277.AD-1285.zip)

_6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_

_7. CvMiniDump.dmp File (ATTACH IF REPORTING A GAME CRASH)_

_8. Steps to reproduce the Issue (Optional)_

_9. Screenshots of the Issue (Optional)_
![Screenshot (119)](https://github.com/user-attachments/assets/44f09dcf-a2f3-48be-b852-6a821fb37bb3)

**Comment by Hokath** [comment]
Nice one

**Comment by saamohod** [comment]
I think it's related to the fresh bug when a scout can pass through a unit of another civilization and grab an ancient ruin on which that unit was sitting.
At the start of the game I often employ a somewhat cheesy tactic (cheesy because afaik AI doesn't use it) to explore a bit with my first warrior and then place that warrior over a goody hut before my scout arrives.
So a couple of times a rival scout managed to snatch that ruin literally from under my warriors feet.

**Comment by axatin** [comment]
no, the bug here is much older. It's a bug in checking which moves are valid. A land unit shouldn't be allowed to embark to a plot with an enemy civilian, but that restriction is not applied when ordering a unit to move to a different water plot. The screenshot shows it quite clearly.
